loader/loader.go

```diff
@@ -529,7 +529,44 @@ func (l *Loader) Init(ctx context.Context) (err error) {
 		}
 	}
 	if !hasSQLMode {
-		lcfg.To.Session["sql_mode"] = l.cfg.LoaderConfig.SQLMode
+		// When upstream's datatime is 2020-00-00, 2020-00-01, 2020-06-00
+		// and so on, downstream will be 2019-11-30, 2019-12-01, 2020-05-31,
+		// as if set the 'NO_ZERO_IN_DATE', 'NO_ZERO_DATE'.
+		// This is because the implementation of go-mysql, that you can see
+		// https://github.com/go-mysql-org/go-mysql/blob/master/replication/row_event.go#L1063-L1087
+		sqlModes := strings.Split(l.cfg.LoaderConfig.SQLMode, ",")
+		needDisable := []string{
+			"NO_ZERO_IN_DATE",
+			"NO_ZERO_DATE",
+			"ERROR_FOR_DIVISION_BY_ZERO",
+			"NO_AUTO_CREATE_USER",
+			"STRICT_TRANS_TABLES",
+			"STRICT_ALL_TABLES",
+		}
+		for _, disable := range needDisable {
+			for i, sqlMode := range sqlModes {
+				if disable == sqlMode {
+					sqlModes = append(sqlModes[:i], sqlModes[i+1:]...)
+					break
+				}
+			}
+		}
+		needEnable := []string{
+			"IGNORE_SPACE",
+			"NO_AUTO_VALUE_ON_ZERO",
+			"ALLOW_INVALID_DATES",
+		}
+		waitEnable := sqlModes
+		for _, enable := range needEnable {
+			for _, sqlMode := range sqlModes {
+				if enable != sqlMode {
+					waitEnable = append(waitEnable, enable)
+					break
+				}
+			}
+		}
```

Comment:
How about extracting the same logic to a function. We can put it in pkg/utils

Reply:
No problem. Do we need to do this whenever there are two duplicate logics?